src/compiler/utilities.ts

```diff
@@ -10068,16 +10111,15 @@ export const emptyFileSystemEntries: FileSystemEntries = {
  *
  * @internal
  */
-export function matchPatternOrExact(patternOrStrings: readonly (string | Pattern)[], candidate: string): string | Pattern | undefined {
-    const patterns: Pattern[] = [];
-    for (const patternOrString of patternOrStrings) {
-        if (patternOrString === candidate) {
-            return candidate;
-        }
+export function matchPatternOrExact(patternOrStrings: ParsedPatterns, candidate: string): string | Pattern | undefined {
+    const { matchableStringSet, patterns } = patternOrStrings;

```

Comment:
```suggestion
    const { matchableStringSet, patterns } = parsedPatterns;

```
just because, well, it's not an array of pattern or string anymore. The comment above this function also needs updating.